src/rabbit_error_logger_file_h.erl

```diff
@@ -65,18 +80,51 @@ init(File) ->
 
 init_file(File, {error_logger, Buf}) ->
     case init_file(File, error_logger) of
-        {ok, {Fd, File, PrevHandler}} ->
-            [handle_event(Event, {Fd, File, PrevHandler}) ||
+        {ok, State} ->
+            [handle_event(Event, State) ||
                 {_, Event} <- lists:reverse(Buf)],
-            {ok, {Fd, File, PrevHandler}};
+            {ok, State};
         Error ->
             Error
     end;
 init_file(File, PrevHandler) ->
     process_flag(trap_exit, true),
     case file:open(File, [append]) of
-        {ok,Fd} -> {ok, {Fd, File, PrevHandler}};
-        Error   -> Error
+        {ok, Fd} ->
+            State =
+                case otp_release_181_or_newer() of
+                    true ->
+                        #st{fd           = Fd,
+                            filename     = File,
+                            prev_handler = PrevHandler,
+                            depth        = get_depth()};
+                    _    ->
+                        {Fd, File, PrevHandler}
+                end,
+            {ok, State};
+        Error    -> Error
+    end.
+
+%% OTP 18.1 introduced the new #st record.
+%% TODO we should use a proper Semver library.
+otp_release_181_or_newer() ->
```

Comment:
I feel I need to add this here

![http://i.imgur.com/Qr8cyzp.jpg](http://i.imgur.com/Qr8cyzp.jpg)